index.js

```diff
@@ -94,11 +94,25 @@ const oauthPlugin = fp(function (fastify, options, next) {
     getAccessTokenFromAuthorizationCodeFlowCallbacked(request, callback)
   }
 
+  function getNewAccessTokenWithRefreshTokenCallbacked (refreshToken, params, callback) {
+    const accessToken = fastify[name].accessToken.create({ refresh_token: refreshToken })
+    accessToken.refresh(params, callback)
+  }
+  const getNewAccessTokenWithRefreshTokenPromisified = promisify(getNewAccessTokenWithRefreshTokenCallbacked)
+
+  function getNewAccessTokenWithRefreshToken (refreshToken, params, callback) {
+    if (!callback) {
+      return getNewAccessTokenWithRefreshTokenPromisified(refreshToken, params)
+    }
+    getNewAccessTokenWithRefreshTokenCallbacked(refreshToken, params, callback)
+  }
+
   const oauth2 = oauth2Module.create(credentials)
 
   if (startRedirectPath) {
     fastify.get(startRedirectPath, startRedirectHandler)
     fastify.decorate('getAccessTokenFromAuthorizationCodeFlow', getAccessTokenFromAuthorizationCodeFlow)
+    fastify.decorate('getNewAccessTokenWithRefreshToken', getNewAccessTokenWithRefreshToken)
```

Comment:
`getNewAccessTokenWithRefreshToken` => `getNewAccessTokenFromRefreshToken` ?

Reply:
Maybe "using" is even more accurate. What do you think?

Reply:
Go!

Reply:
Updated and also added some more docs explaining the role of the 2 decorators in the README.